Sources/AuthFoundation/Migration/Migrators/OIDCLegacyMigrator.swift

```diff
@@ -0,0 +1,270 @@
+            archiver.setClass(StateManager.self, forClassName: "OktaOidc.OktaOidcStateManager")
+            archiver.setClass(StateManager.AuthState.self, forClassName: "OKTAuthState")
+            archiver.setClass(StateManager.AuthState.self, forClassName: "OIDAuthState")
+            archiver.setClass(StateManager.TokenResponse.self, forClassName: "OKTTokenResponse")
+            archiver.setClass(StateManager.TokenResponse.self, forClassName: "OIDTokenResponse")
+            archiver.setClass(StateManager.AuthorizationResponse.self, forClassName: "OKTAuthorizationResponse")
+            archiver.setClass(StateManager.AuthorizationResponse.self, forClassName: "OIDAuthorizationResponse")
+            
+            defer { archiver.finishDecoding() }
+
+            return try archiver.decodeTopLevelObject(of: StateManager.self,
+                                                     forKey: NSKeyedArchiveRootObjectKey)
+        }
+        
+        private func importToken(_ item: Keychain.Search.Result, from model: StateManager?) throws {
+            guard let tokenResponse = model?.authState?.lastTokenResponse,
+                  let tokenType = tokenResponse.tokenType,
+                  let expiresIn = tokenResponse.accessTokenExpirationDate?.timeIntervalSinceNow,
+                  let accessToken = tokenResponse.accessToken,
+                  let scope = tokenResponse.scope
+            else {
+                return
+            }
+            
+            let idToken: JWT?
+            if let idTokenString = tokenResponse.idToken {
+                idToken = try JWT(idTokenString)
+            } else {
+                idToken = nil
+            }
+
+            let configuration = OAuth2Client.Configuration(baseURL: issuer,
+                                                           clientId: clientId,
+                                                           scopes: scopes)
+            let clientSettings: [String: String] = [
+                "client_id": clientId,
+                "redirect_uri": redirectUri.absoluteString,
+                "scope": scope
+            ]
+            
+            let issueDate = idToken?.issuedAt ?? Date()
+
+            let token = Token(id: item.account,
+                              issuedAt: issueDate,
+                              tokenType: tokenType,
+                              expiresIn: expiresIn,
+                              accessToken: accessToken,
+                              scope: scope,
+                              refreshToken: tokenResponse.refreshToken,
+                              idToken: idToken,
+                              deviceSecret: nil,
+                              context: Token.Context(configuration: configuration,
+                                                     clientSettings: clientSettings))
+            
+            var security = Credential.Security.standard
+            if let accessibility = item.accessibility {
```

Comment:
Preserving the keychain accessibility / access control settings from the previous keychain.